### ISSUE TITLE
ZeroHead与YOLOv5Head有什么区别

### ISSUE THREAD

**Issue by violet2422** [tinyvision/DAMO-YOLO#40]
### Before Asking

- [X] I have read the [README](https://github.com/tinyvision/DAMO-YOLO/blob/master/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [X] I want to train my custom dataset, and I have read the [tutorials for finetune on your data](https://github.com/tinyvision/DAMO-YOLO/blob/master/README.md#Finetune%20on%20your%20data) carefully and organize my dataset correctly; 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the DAMO-YOLO [issues](https://github.com/tinyvision/DAMO-YOLO/issues) and found no similar questions.

### Question

很赞赏DAMO-YOLO这相崭新的工作，请问ZeroHead与YOLOv5的Head在网络结构上有什么不同，是否很相似？
ZeroHead：https://github.com/tinyvision/DAMO-YOLO/blob/master/damo/base_models/heads/zero_head.py#L146-L167
```python
 self.cls_convs = nn.ModuleList()
  self.reg_convs = nn.ModuleList()

  for i in range(len(self.strides)):
      cls_convs, reg_convs = self._build_not_shared_convs(
          self.in_channels[i], self.feat_channels[i])
      self.cls_convs.append(cls_convs)
      self.reg_convs.append(reg_convs)

  self.gfl_cls = nn.ModuleList([
      nn.Conv2d(self.feat_channels[i],
                self.cls_out_channels,
                3,
                padding=1) for i in range(len(self.strides))
  ])

  self.gfl_reg = nn.ModuleList([
      nn.Conv2d(self.feat_channels[i],
                4 * (self.reg_max + 1),
                3,
                padding=1) for i in range(len(self.strides))
  ])
```
YOLOv5 Head:  https://github.com/ultralytics/yolov5/blob/master/models/yolo.py#L53
`self.m = nn.ModuleList(nn.Conv2d(x, self.no * self.na, 1) for x in ch)  # output conv`
其中的检测头类：
```python
class Detect(nn.Module):
    stride = None  # strides computed during build
    onnx_dynamic = False  # ONNX export parameter
    export = False  # export mode

    def __init__(self, nc=80, anchors=(), ch=(), inplace=True):  # detection layer
        super().__init__()
        self.nc = nc  # number of classes
        self.no = nc + 5  # number of outputs per anchor
        self.nl = len(anchors)  # number of detection layers
        self.na = len(anchors[0]) // 2  # number of anchors
        self.grid = [torch.zeros(1)] * self.nl  # init grid
        self.anchor_grid = [torch.zeros(1)] * self.nl  # init anchor grid
        self.register_buffer('anchors', torch.tensor(anchors).float().view(self.nl, -1, 2))  # shape(nl,na,2)
        self.m = nn.ModuleList(nn.Conv2d(x, self.no * self.na, 1) for x in ch)  # output conv
        self.inplace = inplace  # use inplace ops (e.g. slice assignment)
```

**Comment by cwhgn** [comment]
感谢关注。我们的head和yolov5很类似，只有一个线性层来进行分类和回归。我们设计成这样，主要是因为我们发现把计算量放到neck能带来更多的收益，如下表所示。另外值得一提的是，我们head中回归点的含义和yolov5不太一样，yolov5回归的是检测框的中心点偏移和wh。我们回归的是bbox边界的分布，具体可以参考[gfocal loss](https://arxiv.org/abs/2006.04388)。
<td><img src="https://user-images.githubusercontent.com/6065208/207015570-b23cbd48-992d-4623-96d4-90cd074d44a7.png" width=400px></td>
